docs/built-in-functions.rst

```diff
@@ -3,176 +3,309 @@
 
-    Returns a value which is the default (zeroed) value of its type.
+.. py:function:: bitwise_or(x: uint256, y: uint256) -> uint256
 
-    * ``typename``: Name of the type
+    Perform a "bitwise or" operation. Each bit of the output is 0 if the corresponding bit of ``x`` AND of ``y`` is 0, otherwise it's 1.
 
-    For instance, `xs: uint256[5] = empty(uint256[5])`
+    .. code-block:: python
 
-.. py:function:: as_wei_value(value: int, unit: str) -> uint256
+        @external
+        @view
+        def foo(x: uint256, y: uint256) -> uint256:
+            return bitwise_or(x, y)
 
-    Takes an amount of ether currency specified by a number and a unit and returns the integer quantity of wei equivalent to that amount.
+    .. code-block:: python
 
-    * ``value``: Value for the ether unit
-    * ``unit``: Ether unit name (e.g. ``"wei"``, ``"ether"``, ``"gwei"``, etc.) indicating the denomination of ``value``.
+        >>> ExampleContract.foo(31337, 8008135)
+        8027119
 
-.. py:function:: slice(b: Union[Bytes, bytes32, String], start: uint256, length: uint256) -> Union[Bytes, String]
+.. py:function:: bitwise_xor(x: uint256, y: uint256) -> uint256
 
-    Copy a list of bytes and return a specified slice.
+    Perform a "bitwise exclusive or" operation. Each bit of the output is the same as the corresponding bit in ``x`` if that bit in ``y`` is 0, and it's the complement of the bit in ``x`` if that bit in ``y`` is 1.
 
-    * ``b``: value being sliced
-    * ``start``: start position of the slice
-    * ``length``: length of the slice
+    .. code-block:: python
 
-    If the value being sliced is a ``Bytes`` or ``bytes32``, the return type is ``Bytes``.  If it is a ``String``, the return type is ``String``.
+        @external
+        @view
+        def foo(x: uint256, y: uint256) -> uint256:
+            return bitwise_xor(x, y)
+
+    .. code-block:: python
+
+        >>> ExampleContract.foo(31337, 8008135)
+        8014766
+
+.. py:function:: shift(x: uint256, _shift: int128) -> uint256
+
+    Return ``x`` with the bits shifted ``_shift`` places. A positive ``_shift`` value equals a left shift, a negative value is a right shift.
 
     .. code-block:: python
 
         @external
         @view
-        def foo(s: string[32]) -> string[5]:
-            return slice(s, 4, 5)
+        def foo(x: uint256, y: int128) -> uint256:
+            return shift(x, y)
 
     .. code-block:: python
 
-        >>> ExampleContract.foo("why hello! how are you?")
-        "hello"
+        >>> ExampleContract.foo(2, 8)
+        512
 
-.. py:function:: len(b: Union[Bytes, String]) -> uint256
+Chain Interaction
+=================
 
-    Return the length of a given ``Bytes`` or ``String``.
+.. py:function:: create_forwarder_to(target: address, value: uint256 = 0) -> address
+
+    Duplicates a contract's code and deploys it as a new instance, by means of a ``DELEGATECALL``.
```

Comment:
Doesn't actually duplicate lol. It creates a forwarder contract that delegates all logic to a given target template contract

Reply:
I just moved the text that was already written.  Can you suggest an edit?

Reply:
```suggestion
    Creates a small contract and deploys it as a new instance of the contract at ``target``, which duplicates the logic of ``target`` but has it's own state since every call is made using ``DELEGATECALL`` to ``target``.
```